src/test/java/io/zksync/IntegrationTestTransactionReceiptProcessor.java

```diff
@@ -0,0 +1,257 @@
+package io.zksync;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.security.SecureRandom;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.web3j.protocol.Web3j;
+import org.web3j.protocol.core.methods.response.TransactionReceipt;
+import org.web3j.protocol.http.HttpService;
+import org.web3j.tuples.generated.Tuple2;
+import org.web3j.tx.gas.DefaultGasProvider;
+import org.web3j.utils.Convert;
+import org.web3j.utils.Numeric;
+import org.web3j.utils.Convert.Unit;
+
+import io.zksync.domain.ChainId;
+import io.zksync.domain.TransactionBuildHelper;
+import io.zksync.domain.auth.ChangePubKeyECDSA;
+import io.zksync.domain.fee.TransactionFeeBatchRequest;
+import io.zksync.domain.fee.TransactionFeeDetails;
+import io.zksync.domain.fee.TransactionType;
+import io.zksync.domain.swap.Order;
+import io.zksync.domain.token.NFT;
+import io.zksync.domain.token.Token;
+import io.zksync.domain.transaction.ChangePubKey;
+import io.zksync.domain.transaction.ForcedExit;
+import io.zksync.domain.transaction.MintNFT;
+import io.zksync.domain.transaction.Swap;
+import io.zksync.domain.transaction.TransactionDetails;
+import io.zksync.domain.transaction.Transfer;
+import io.zksync.domain.transaction.Withdraw;
+import io.zksync.domain.transaction.WithdrawNFT;
+import io.zksync.ethereum.EthereumProvider;
+import io.zksync.provider.AsyncProvider;
+import io.zksync.signer.DefaultEthSigner;
+import io.zksync.signer.EthSignature;
+import io.zksync.signer.ZkSigner;
+import io.zksync.transport.ZkTransactionStatus;
+import io.zksync.transport.receipt.ZkSyncPollingTransactionReceiptProcessor;
+import io.zksync.transport.receipt.ZkSyncTransactionReceiptProcessor;
+import io.zksync.wallet.SignedTransaction;
+import io.zksync.wallet.ZkASyncWallet;
+
+@Ignore
+public class IntegrationTestTransactionReceiptProcessor {
```

Comment:
These tests should completely replace the `IntegrationTestFullFlow`. The ReceiptProcessor should be used there whenever possible